skfda/exploratory/visualization/_ddplot.py

```diff
@@ -0,0 +1,142 @@
+"""DD-Plot Module.
+
+This module contains the necessary functions to construct the DD-Plot.
+To do this depth is calculated for the two chosen distributions, and then
+a scatter plot is created of this two variables.
+"""
+
+from typing import Callable
+from ... import FDataGrid
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from ._utils import (
+    _get_figure_and_axes,
+    _set_figure_layout_for_fdata,
+)
+
+
+class DDPlot:
+
+    """DDPlot visualization. Plot the depth of our fdata elements in two
+    different distributions, one in each axis. It is useful to understand
+    how our data is more related with one subset of data / distribution
+    than another one.
+
+    Args:
+        fdata: functional data set that we want to examine.
+        dist1: functional data set that represents the first distribution that
+            we want to use to compute the depth (Depth X).
+        dist2: functional data set that represents the second distribution that
+            we want to use to compute the depth (Depth Y).
+        depth_method: method that will be used to compute the depths of the
+            data with respect to the distributions.
+
+    """
+
+    def __init__(
+        self,
+        fdata: FDataGrid,
```

Comment:
The types for all of them are `T` a TypeVar object.

Reply:
ok, so fdata, dist1, dist2 and depth_method are T? Sorry, i didnt understand which ones you mean

Reply:
`fdata`, `dist1` and `dist2` are a `TypeVar` object (call it `T` or maybe a more meaningful name), and `depth_method` is then `Depth[T]`. That would allow, for example, calling `DDPlot` with multivariate data and a multivariate depth, or with functional data and a functional depth.